File: Z.ExtensionMethods-master/src/Z.Reflection/System.Reflection.ParameterInfo/System.Attribute/ParameterInfo.IsDefined.cs
| Enhance the .NET Framework and .NET Core with over 1000 extension methods.


():


using System;
using System.Reflection;

public static partial class Extensions
{
    /// <summary>
    ///     Determines whether any custom attributes are applied to a method parameter. Parameters specify the method
    ///     parameter, and the type of the custom attribute to search for.
    /// </summary>
    /// <param name="element">An object derived from the  class that describes a parameter of a member of a class.</param>
    /// <param name="attributeType">The type, or a base type, of the custom attribute to search for.</param>
    /// <returns>true if a custom attribute of type  is applied to ; otherwise, false.</returns>
    public static Boolean IsDefined(this ParameterInfo element, Type attributeType)
    {
        return Attribute.IsDefined(element, attributeType);
    }

    /// <summary>
    ///     Determines whether any custom attributes are applied to a method parameter. Parameters specify the method
    ///     parameter, the type of the custom attribute to search for, and whether to search ancestors of the method
    ///     parameter.
    /// </summary>
    /// <param name="element">An object derived from the  class that describes a parameter of a member of a class.</param>
    /// <param name="attributeType">The type, or a base type, of the custom attribute to search for.</param>
    /// <param name="inherit">If true, specifies to also search the ancestors of  for custom attributes.</param>
    /// <returns>true if a custom attribute of type  is applied to ; otherwise, false.</returns>
    public static Boolean IsDefined(this ParameterInfo element, Type attributeType, Boolean inherit)
    {
        return Attribute.IsDefined(element, attributeType, inherit);
    }
}

File: Z.ExtensionMethods-master/src (VB.NET)/Z.ExtensionMethods.VB/Z.Reflection/System.Reflection.ParameterInfo/System.Attribute/ParameterInfo.IsDefined.vb

' Copyright (c) 2015 ZZZ Projects. All rights reserved
' Licensed under MIT License (MIT) (https://github.com/zzzprojects/Z.ExtensionMethods)
' Website: http://www.zzzprojects.com/
' Feedback / Feature Requests / Issues : http://zzzprojects.uservoice.com/forums/283927
' All ZZZ Projects products: Entity Framework Extensions / Bulk Operations / Extension Methods /Icon Library

Imports System.Reflection

Public Module Extensions_849

	''' <summary>
	'''     Determines whether any custom attributes are applied to a method parameter. Parameters specify the method
	'''     parameter, and the type of the custom attribute to search for.
	''' </summary>
	''' <param name="element">An object derived from the  class that describes a parameter of a member of a class.</param>
	''' <param name="attributeType">The type, or a base type, of the custom attribute to search for.</param>
	''' <returns>true if a custom attribute of type  is applied to ; otherwise, false.</returns>
	<System.Runtime.CompilerServices.Extension> _
	Public Function IsDefined(element As ParameterInfo, attributeType As Type) As [Boolean]
		Return Attribute.IsDefined(element, attributeType)
	End Function

	''' <summary>
	'''     Determines whether any custom attributes are applied to a method parameter. Parameters specify the method
	'''     parameter, the type of the custom attribute to search for, and whether to search ancestors of the method
	'''     parameter.
	''' </summary>
	''' <param name="element">An object derived from the  class that describes a parameter of a member of a class.</param>
	''' <param name="attributeType">The type, or a base type, of the custom attribute to search for.</param>
	''' <param name="inherit">If true, specifies to also search the ancestors of  for custom attributes.</param>
	''' <returns>true if a custom attribute of type  is applied to ; otherwise, false.</returns>
	<System.Runtime.CompilerServices.Extension> _
	Public Function IsDefined(element As ParameterInfo, attributeType As Type, inherit As [Boolean]) As [Boolean]
		Return Attribute.IsDefined(element, attributeType, inherit)
	End Function
End Module


